ARCHITECTURE.md

```diff
@@ -1,6 +1,266 @@
+
+```ts
+const wf = workflowFactory
+  // Registers an action with the workflow using the action class directly
+  .registerAction(MultiSenderAction) // => new MultiSenderAction(options)
+  // Registers a trigger with the workflow using the trigger class directly
+  .registerTrigger(PriceBasedTrigger) // => new PriceBasedTrigger(options, commonConfig)
+```
+
+
+### Using Configuration
+
+Below is a complete example of a workflow configuration:
+
+```ts
+const workflowFactory = new WorkflowsFactory(provider);
+
+const config = {
+  name: 'Send assets to multiple addresses',
+  issuer: 'Ditto',
+  hint: 'Send your assets to multiple addresses in one transaction',
+  triggers: [
+    {
+      type: 'Core.Scheduled',
+      options: {
+        frequency: 'daily',
+        time: '08:00'
+      }
+    }
+  ],
+  actions: [
+    {
+      type: 'Application.MultiSender',
+      options: {
+        items: [
+          {
+            asset: {
+              address: '0xTokenAddress',
+              decimals: 18
+            },
+            amount: '100',
+            to: '0xRecipientAddress'
+          },
+          {
+            asset: null,
+            amount: '0.1',
+            to: '0xAnotherRecipientAddress'
+          }
+        ]
+      }
+    }
+  ],
+};
+
+const wf = workflowFactory
+  .registerAction(MultiSenderAction)
+  .registerTrigger(ScheduledTrigger)
+  // Creates a workflow from the given configuration object.
+  .createFromConfig(config);
+
+const deployedWorkflow = await wf.buildAndDeploy(swAddress, accountAddress);
+```
+
+This setup is equivalent to the following:
+
+```ts
+const workflowFactory = new WorkflowsFactory(provider);
+
+const wf = await workflowFactory.create({
+  name: 'My first workflow',
+  triggers: [
+    new ScheduledTrigger(
+      {
+        frequency: 'daily',
+        time: '08:00'
+      },
+      commonConfig
+    ),
+  ],
+  actions: [
+    new MultiSenderActionCallDataBuilder(
+      {
+        items: [
+          {
+            asset: {
+              address: '0xTokenAddress',
+              decimals: 18
+            },
+            amount: '100',
+            to: '0xRecipientAddress'
+          },
+          {
+            asset: null,
+            amount: '0.1',
+            to: '0xAnotherRecipientAddress'
+          }
+        ]
+      },
+      commonConfig
+    ),
+  ],
+  chainId,
+});
+
+const deployedWorkflow = await wf.buildAndDeploy(swAddress, accountAddress);
+```
+
+
+### Implementing Actions
+
+Actions should implement the `AbstractAction` class with a static `id` and a `build` method.
+
+```ts
+abstract class AbstractAction {
+  static id: string;
+
+  abstract build(): Promise<CallDataBuilderReturnData>;
+}
+```
+
+Each action class should define a unique static `id` to identify the action type and implement the `build` method to return the necessary data for the action.
+
+```ts
+class MultiSenderAction extends AbstractAction {
+  static id = 'Application.MultiSender';
```

Comment:
As for the ID, we may have problems with that. We want to be able to create custom triggers and actions, which means that there may be collisions with the identification of actions & triggers. In addition, it is not fully disclosed how it will be serialized and deserialized.